docs/RFCS/20201020_sql_query_denylist.md

```diff
@@ -0,0 +1,183 @@
+- Feature Name: SQL Query Denylist
+- Status: draft
+- Start Date: 2020-10-20
+- Authors: angelapwen, otan
+- RFC PR: (PR # after acceptance of initial draft)
+- Cockroach Issue: [51643](https://github.com/cockroachdb/cockroach/issues/51643)
+
+# Summary
+
+We propose a mechanism built into CockroachDB to prevent operator-defined SQL queries from being executed. This will immediately address the situations where clusters enter unresponsive states due to a query, or a query pattern, that has been identified. To bring the clusters back up as soon as possible and prevent them from going down again due to the same query, we need a query denylist on the database. This RFC details a file-managed SQL query denylist implementation that will allow database administrators to trigger query denial by uploading a YAML file comprising regular expression patterns to a customizable file location.  
+# Motivation
+
+It is important to support a file-managed denylist so that database administrators are able to quickly deny queries that have been confirmed to bring down clusters to the point of unresponsiveness. In this situation, SQL commands cannot be run from the console and cluster settings will not be able to be set. 
+
+Without this functionality, there may be applications endlessly retrying the “bad” queries that will continuously bring the cluster down when executed. There is currently no built-in way to stop execution of a query like this other than by preventing the query from being run in the first place, and it is time- and resource-intensive for developers to write their own. 
```

Comment:
kind of a nit, but i'd use a word other than retry here. retries make me think of a query being retried because the previous time failed. but we also are more concerned with the application generating more of a certain type of query in the process of handling more user requests.

Reply:
Yeah, that makes sense. I hadn't thought about how the word retry is overloaded in the context of application code vs. our database interns.